src/ktopt_planning_context.cpp

```diff
@@ -31,15 +33,31 @@ void KTOptPlanningContext::solve(planning_interface::MotionPlanDetailedResponse&
 
 void KTOptPlanningContext::solve(planning_interface::MotionPlanResponse& res)
 {
+  RCLCPP_INFO(getLogger(),
+              "Setting up and solving optimization problem ..)"
+              );
   // preliminary house keeping
   const auto time_start = std::chrono::steady_clock::now();
   res.planner_id = std::string("ktopt");
   res.error_code.val = moveit_msgs::msg::MoveItErrorCodes::SUCCESS;
 
   // Retrieve motion plan request
   const auto& req = getMotionPlanRequest();
-  const moveit::core::RobotState start_state(*getPlanningScene()->getCurrentStateUpdated(req.start_state));
-  // TODO: update plant_ state here as well
+  const moveit::core::RobotState start_state(
+    *getPlanningScene()->getCurrentStateUpdated(
+      req.start_state));
+  const auto group = getPlanningScene()->getRobotModel()->getJointModelGroup(
+    getGroupName());
+  RCLCPP_INFO_STREAM(getLogger(),
+                     "Planning for group"
+                     << getGroupName());
+  const auto& joints = group->getActiveJointModels();
+
+  // q represents the complete state (joint positions and velocities)
+  const auto q = moveit_to_drake_complete_state(start_state, joints);
+
+  // drake accepts a VectorX<T> 
+  plant_->SetPositionsAndVelocities(plant_context_, q);
```

Comment:
Do you need to call `SetPositionsAndVelocities`, or can you just call `SetPositions` on the plant, thereby doing away with the `moveit_to_drake_complete_state()` function?

Reply:
Fair, I just thought `SetPositionsAndVelocities` is neater from "constructing and Optimization problem" perspective, I am not sure what the default values are within the plant. I'll change `moveit_to_drake_complete_state` function to reflect what it is actually doing here.